src/Lifecycle.ts

```diff
@@ -761,28 +770,40 @@ async function showStorageEvictedDialog(): Promise<boolean> {
 // `instanceof`. Babel 7 supports this natively in their class handling.
 class AbortLoginAndRebuildStorage extends Error {}
 
-async function persistCredentials(credentials: IMatrixClientCreds): Promise<void> {
-    localStorage.setItem(HOMESERVER_URL_KEY, credentials.homeserverUrl);
-    if (credentials.identityServerUrl) {
-        localStorage.setItem(ID_SERVER_URL_KEY, credentials.identityServerUrl);
-    }
-    localStorage.setItem("mx_user_id", credentials.userId);
-    localStorage.setItem("mx_is_guest", JSON.stringify(credentials.guest));
-
-    // store whether we expect to find an access token, to detect the case
+/**
+ * Persist a token in storage
+ * When pickle key is present, will attempt to encrypt the token
+ * Stores in idb, falling back to localStorage
+ *
+ * @param storageKey key used to store the token
+ * @param name eg "access_token" used as initialization vector during encryption
+ * @param token
+ * @param pickleKey optional pickle key used to encrypt token
+ */
+async function persistTokenInStorage(
+    storageKey: string,
+    name: string,
+    token: string | undefined,
+    pickleKey: IMatrixClientCreds["pickleKey"],
```

Comment:
```suggestion
    pickleKey: string | undefined,
```

Reply:
Why is that better?

Reply:
so that, if I want to call this function, I don't have to go and look up wtf an `IMatrixClientCreds["pickleKey"]` is before I know what type I can pass here.